dali/kernels/signal/fft/fft_cpu.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/signal/fft/fft_cpu.h"
+#include "dali/kernels/signal/fft/fft_cpu_impl_ffts.h"
+#include <ffts/ffts.h>
+#include <cmath>
+#include <complex>
+#include "dali/core/common.h"
+#include "dali/core/convert.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/format.h"
+#include "dali/core/util.h"
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+namespace fft {
+
+template <typename OutputType, typename InputType, int Dims>
+KernelRequirements Fft1DCpu<OutputType, InputType, Dims>::Setup(
+    KernelContext &context,
+    const InTensorCPU<InputType, Dims> &in,
+    const FftArgs &args) {
+  if (!impl_ || args != args_) {
+    impl_.reset(new impl::Fft1DImplFfts<OutputType, InputType, Dims>());
```

Comment:
Why not `make_unique`?

Reply:
will change

Reply:
NOTE: If you're creating the sole instance of this object in .cc file, it's better to have the destructor of Fft1DCpu defined in .cc file, too - this will make sure that the memory is freed by exactly the same deleter when using the class across shared object boundary.

Reply:
done